corehq/apps/user_importer/importer.py

```diff
@@ -389,11 +391,37 @@ def create_or_update_users_and_groups(domain, user_specs, group_memoizer=None, u
                         user.reset_locations(location_ids, commit=False)
 
                 if role:
-                    user.set_role(domain, roles_by_name[role].get_qualified_id())
+                    role = roles_by_name[role].get_qualified_id()
+                    user.set_role(domain, role)
+
+                if web_user:
+                    user.user_data.update({'login_as_user': web_user})
 
                 user.save()
 
-                if send_account_confirmation_email:
+                if web_user:
+                    current_user = CouchUser.get_by_username(web_user)
+                    if not current_user and is_account_confirmed:
+                        raise UserUploadError(_(
+                            f"You can only set 'Is Account Confirmed' to 'True' on an existing Web User. {web_user} is a new username."
+                        ))
+                    if current_user and not current_user.is_member_of(domain) and is_account_confirmed:
```

Comment:
nit: I don't have a great suggestion, but it'd be nice if these conditionals were a bit more readable. could make named variables or comments.